File: Laba1/Laba1/oil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba1
{
    class oil
    {
        enum state{solid,liquid}

        state State = state.solid;

        private int tempetature = 0;

        public bool isReady { get { if (State == state.liquid) return true; else return false; } }

        public void GetHeat()
        {
            if (tempetature < 80)
            {
                tempetature += 10;
            }
            if(tempetature >= 80)State = state.liquid;
        }
    }
}

File: Laba1/src/main/Oil.java
package main;

public class Oil {

    enum state{solid,liquid};

    private state State=state.solid;

    private int temperature=0;

    public boolean isReady(){
        return (State==state.liquid)?true:false;
    }

    public void GetHeat(){
        if(temperature<80){
            temperature+=10;
        }
        if(temperature>=80)State=state.liquid;
    }

}
